packages/dds/tree/src/feature-libraries/object-forest/objectForest.ts

```diff
@@ -398,18 +398,25 @@ class Cursor implements ITreeSubscriptionCursor {
         return { result: TreeNavigationResult.NotFound, moved: 0 };
     }
     up(): TreeNavigationResult {
-        if (this.parentStack.pop() === undefined) {
-            // We are at the root, so return NotFound without making any changes to the state.
+        const length = this.parentStack.length;
+        assert(this.indexStack.length === length, "Unexpected indexStack.length");
+        assert(this.keyStack.length === length - 1, "Unexpected keyStack.length");
+
+        // Already at the root
```

Comment:
It took me a little time to understand why we're already at the root even though there's a parent, could you add a quick explanation here?

Reply:
comments added.